docs: Rewrite and expand Fleet Manager API endpoints documentation (#1696)
bf2fc6cc711aee1a0c2a/kas-fleet-manager: docs/admin-api-overview.md

# Admin API overview

Fleet Manager provides a set of administrative endpoints which are a set of
privileged endpoints only intended to be used by administrator of the Fleet
Manager. The reason for that is that these endpoints allow
performing potentially destructive actions, actions that should only be
performed by Fleet Manager administrators, accessing sensitive information
and accessing cross-tenant information. This set of endpoints are also often
referred as Admin API endpoints.

Authentication and Authorization is performed for the requests made against
Admin API endpoints. OIDC is the protocol used to perform Authentication
and Authorization. Therefore, when making a request to the Admin API endpoints

> an OIDC token has to be provided along it.

Reviewer: ```suggestion
, an OIDC token has to be provided with such request.
```

The available Fleet Manager API endpoints are documented in:
* [KAS Fleet Manager Admin API OpenAPI specification file](../openapi/kas-fleet-manager-private-admin.yaml)
* [Connector Fleet Manager Admin API OpenAPI specifications file](../openapi/connector_mgmt-private-admin.yaml)

## Authentication

> To be able to perform Authentication in the Admin API endpoints the location of

Author: updated

the OIDC authorization server that issues the OIDC tokens used to call
the Fleet Manager Admin API endpoints has to be provided. This is configured
through the following Fleet Manager CLI flags:
* `--admin-api-sso-base-url`: The Base URL of the OIDC authorization server
  endpoint. By default set to `https://auth.redhat.com`
* `--admin-api-sso-endpoint-uri`: The URL path of the OIDC realm in the
  authorization server specified in `--admin-api-sso-base-url`. By default
  set to `/auth/realms/EmployeeIDP`
* `--admin-api-sso-realm`: The name of the OIDC realm to be used in the
  OIDC authorization server endpoint specified in `admin-api-sso-base-url`. By
  default set to `EmployeeIDP`. The value specified here has to match the realm
  name specified as part of the URL path specified in
  `--admin-api-sso-endpoint-uri`

> Specifically, Fleet Manager Manager performs the following as part of the

Reviewer: ```suggestion
Specifically, Fleet Manager performs the following as part of the
```

Author: updated
